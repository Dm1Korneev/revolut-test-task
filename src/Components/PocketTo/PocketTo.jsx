import React from 'react';

import Pocket from 'Components/Pocket';
import CurrentRate from 'Components/CurrentRate';

function PocketTo(props) {
  return (
    <Pocket
      {...props}
      valuePrefix="+"
      cildren={<CurrentRate reverse />}
      inputValueLAbel="Receive amount"
      selectCurrencyLabel="Receive currency"
    />
  );
}

export default PocketTo;
