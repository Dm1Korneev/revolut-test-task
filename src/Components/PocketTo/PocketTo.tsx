import React, { FC } from 'react';

import Pocket, { Props as PocketProps } from 'Components/Pocket';
import CurrentRate from 'Components/CurrentRate';

const PocketTo: FC<PocketProps> = (props: PocketProps): JSX.Element => (
  <Pocket
    {...props}
    valuePrefix="+"
    inputValueLabel="Receive amount"
    selectCurrencyLabel="Receive currency"
  >
    <CurrentRate reverse />
  </Pocket>
);

export default PocketTo;
