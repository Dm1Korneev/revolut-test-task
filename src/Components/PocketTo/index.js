import commonHoc from 'Components/commonHoc';
import { changeReceiveValue, setPocketTo } from 'Redux/actions';
import {
  currenciesSelector, pocketToCurrencySelector, pocketToValueSelector, receiveValueSelector,
} from 'Selectors/session';

import PocketTo from './PocketTo';

const mapStateToProps = (state) => ({
  currency: pocketToCurrencySelector(state),
  valueHave: pocketToValueSelector(state),
  currencies: currenciesSelector(),
  value: receiveValueSelector(state),
});

const mapDispatchToProps = {
  setPocketCurrency: setPocketTo,
  changeValue: changeReceiveValue,
};

export default commonHoc(PocketTo, {
  mapStateToProps,
  mapDispatchToProps,
});
