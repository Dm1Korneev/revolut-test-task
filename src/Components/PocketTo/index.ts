import commonHoc from 'Components/commonHoc';
import { changeReceiveValue, setPocketTo } from 'Redux/actions';
import {
  pocketToCurrencySelector, pocketToValueSelector, receiveValueSelector,
} from 'Selectors/exchange';
import { currenciesSelector } from 'Selectors/currencies';
import { StateType } from 'Redux/store';

import PocketTo from './PocketTo';

const mapStateToProps = (state: StateType): Record<string, any> => ({
  currency: pocketToCurrencySelector(state),
  valueHave: pocketToValueSelector(state),
  currencies: currenciesSelector(),
  value: receiveValueSelector(state),
});

const mapDispatchToProps = {
  setPocketCurrency: setPocketTo,
  changeValue: changeReceiveValue,
};

export default commonHoc(PocketTo, {
  mapStateToProps,
  mapDispatchToProps,
});
