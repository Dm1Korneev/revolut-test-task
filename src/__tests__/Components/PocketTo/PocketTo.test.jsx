import React from 'react';
import { render } from '@testing-library/react';
import PocketTo from 'Components/PocketTo/PocketTo';

jest.mock('Components/Pocket', () => global.mockComponent('Pocket'));
jest.mock('Components/CurrentRate', () => global.mockComponent('CurrentRate'));

describe('PocketTo component', () => {
  const props = {
    test: 'test',
  };

  test('renders the component', () => {
    const { asFragment } = render(<PocketTo {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
