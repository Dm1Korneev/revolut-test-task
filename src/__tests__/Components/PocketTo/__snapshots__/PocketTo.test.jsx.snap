// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PocketTo component renders the component 1`] = `
<DocumentFragment>
  <div
    originalcomponent="Pocket"
    test-prop-inputvaluelabel="\\"Receive amount\\""
    test-prop-selectcurrencylabel="\\"Receive currency\\""
    test-prop-test="\\"test\\""
    test-prop-valueprefix="\\"+\\""
  >
    <div
      originalcomponent="CurrentRate"
      test-prop-reverse="true"
    />
  </div>
</DocumentFragment>
`;
